test/extended/util/framework.go

```diff
@@ -189,22 +189,22 @@ var CheckImageStreamTagNotFoundFn = func(i *imageapi.ImageStream) bool {
 func WaitForADeployment(client kclient.ReplicationControllerInterface,
```

Comment:
I think i would have done this differently.... either:

1) just check the start time and calculate and end time, and then if the end time has passed at the top of the outer loop
2) get rid of the the Watch entirely and just wrap the list() in a wait.Poll check (which is basically what WaitForABuild does).

with the existing logic it's pretty hard to follow how long it's really going to end up waiting, based on expiration of watches.  also 3 minutes between polls seems like a lot.